zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/OpenSearchIncidentUpdateRepository.java

```diff
@@ -0,0 +1,516 @@
+  @Override
+  public CompletionStage<Integer> bulkUpdate(final IncidentBulkUpdate bulk) {
+    final var updates = bulk.stream().map(this::createUpdateOperation).toList();
+    final var request =
+        new BulkRequest.Builder()
+            .operations(updates)
+            .source(s -> s.fetch(false))
+            .refresh(Refresh.WaitFor)
+            .build();
+
+    try {
+      return client
+          .bulk(request)
+          .thenComposeAsync(
+              r -> {
+                if (r.errors()) {
+                  return CompletableFuture.failedFuture(collectBulkErrors(r.items()));
+                }
+
+                return CompletableFuture.completedFuture(r.items().size());
+              },
+              executor);
+    } catch (final IOException e) {
+      return CompletableFuture.failedFuture(e);
+    }
+  }
+
+  @Override
+  public CompletionStage<List<String>> analyzeTreePath(final String treePath) {
+    final var request =
+        new AnalyzeRequest.Builder()
+            .field(ListViewTemplate.TREE_PATH)
+            .index(listViewAlias)
+            .text(treePath)
+            .build();
+
+    try {
+      return client
+          .indices()
+          .analyze(request)
+          .thenApplyAsync(
+              response -> response.tokens().stream().map(AnalyzeToken::token).toList(), executor);
+    } catch (final IOException e) {
+      return CompletableFuture.failedFuture(e);
+    }
+  }
+
+  @Override
+  public CompletionStage<Collection<ActiveIncident>> getActiveIncidentsByTreePaths(
+      final Collection<String> treePathTerms) {
+    final var treePathValues = treePathTerms.stream().map(FieldValue::of).toList();
+    final var pathQ =
+        QueryBuilders.terms()
+            .field(IncidentTemplate.TREE_PATH)
+            .terms(v -> v.value(treePathValues))
+            .build()
+            .toQuery();
+    final var stateQ =
+        QueryBuilders.term()
+            .field(IncidentTemplate.STATE)
+            .value(v -> v.stringValue(IncidentState.ACTIVE.name()))
+            .build()
+            .toQuery();
+    final var request =
+        new SearchRequest.Builder()
+            .query(q -> q.bool(b -> b.must(pathQ, stateQ)))
+            .source(s -> s.filter(f -> f.includes(IncidentTemplate.TREE_PATH)))
+            .index(incidentAlias);
+
+    return fetchUnboundedDocumentCollection(
+        request, IncidentEntity.class, h -> new ActiveIncident(h.id(), h.source().getTreePath()));
+  }
+
+  /**
+   * Variant of {@link #fetchUnboundedDocumentCollection(SearchRequest.Builder, Class, Function)} to
+   * use when you don't care about the source document, meaning you won't be using any
+   * deserialization functionality.
+   */
+  private <T> CompletionStage<Collection<T>> fetchUnboundedDocumentCollection(
+      final SearchRequest.Builder requestBuilder, final Function<Hit<Object>, T> transformer) {
+    return fetchUnboundedDocumentCollection(requestBuilder, Object.class, transformer);
```

Comment:
:thought_balloon: there is quite a bit of logic which is very similar (but again not the same because of the types) that could have been abstracted away, but I understand that it may not be worth doing that as well as long as it's tested :+1: